cryostatio/cryostat-helm
# feat(grafana): remove minimal config option, route to Grafana using existing route and service

Based on #118
Depends on #118
Fixes #114

To test:
Install chart on OpenShift with `helm install cryostat --set core.route.enabled=true ./charts/cryostat`, then follow the notes that are printed to set the additional environment variables dependent upon the route host. Wait for the deployment to become ready, then visit the route. Cryostat should be running with no error notifications about Grafana or archives. Create a `localhost:0` custom target, then start a flight recording on Cryostat. You should be able to archive the active recording, generate reports on the active or archived recording, and view the active recording in Grafana. ~~Viewing archived recordings in Grafana is not yet implemented by the Cryostat 3 server but the configuration here should allow that seamlessly once we have that reimplemented.~~

Test along with https://github.com/cryostatio/cryostat3/pull/315 / https://github.com/cryostatio/cryostat3/pull/324 to get more of the UI features working.

~~We might want to keep the `minimal` deployment option (no jfr-datasource and no Grafana dashboard), but I'm leaning toward removing it. Originally this was put in place so that the deployment footprint was only the Cryostat container itself, but nowadays the smallest footprint is Cryostat + database + object storage + auth proxy, and the Grafana dashboard layout has been updated quite a bit to include several more powerful/useful data visualizations, so IMO it's worth making it a standard feature.~~
